controller/project/manager.py

```diff
@@ -219,12 +229,182 @@ def __get_crowd_label_is_data(project_id: str, is_id: str) -> Dict[str, str]:
 
 def __get_first_data_id(project_id: str, user_id: str, huddle_type: str) -> str:
     if huddle_type == enums.LinkTypes.DATA_SLICE.value:
-        slices = ds_manager.get_all(project_id, enums.SliceTypes.STATIC_DEFAULT)
+        slices = data_slice.get_all(project_id, enums.SliceTypes.STATIC_DEFAULT)
         if slices and len(slices) > 0:
             return slices[0].id
     elif huddle_type == enums.LinkTypes.HEURISTIC.value:
-        return information_source_manager.get_first_crowd_is_for_annotator(
-            project_id, user_id
-        )
+        return information_source.get_first_crowd_is_for_annotator(project_id, user_id)
     else:
         raise ValueError("invalid huddle type")
+
+
+# GATES
+projects_updating = set()
+projects_updating_lock = threading.Lock()
+
+
+def get_gates_integration_data(project_id: str) -> GatesIntegrationData:
+
+    project_item = project.get(project_id)
+    if not project_item:
+        raise GraphQLError("Project not found")
+
+    if project_id in projects_updating:
+        return GatesIntegrationData(status=enums.GatesIntegrationStatus.UPDATING.value)
+
+    missing_tokenizer = not __tokenizer_pickle_exists(project_item.tokenizer)
+    missing_embeddings = __get_missing_embedding_pickles(project_id)
+    missing_information_sources = __get_missing_information_source_pickles(project_id)
+
+    status = enums.GatesIntegrationStatus.READY.value
+    if (
+        missing_tokenizer
+        or len(missing_embeddings) > 0
+        or len(missing_information_sources) > 0
+    ):
+        status = enums.GatesIntegrationStatus.NOT_READY.value
+
+    return GatesIntegrationData(
+        status=status,
+        missing_tokenizer=missing_tokenizer,
+        missing_embeddings=missing_embeddings,
+        missing_information_sources=missing_information_sources,
+    )
+
+
+def __tokenizer_pickle_exists(config_string: str) -> bool:
+    tokenizer_path = os.path.join(
+        "/inference/tokenizers", f"tokenizer-{config_string}.pkl"
```

Comment:
is there a specific reason why you are using os.join? i know it should give the correct path seperator for the os but it's fixed to linux in docker anyway (and the path already has some / )

Not wrong just curious if i'd have missed something since i would have gone with `f"/inference/tokenizers/tokenizer-{config_string}.pkl"`